### PR TITLE
Fix: system webhooks API bug

### PR DESCRIPTION
**Pull request by pulltheflower**
- Fix the bug about admin/hooks API that `GET /admin/hooks` can only fetch system_hooks, `POST /admin/hooks` can only create default_hooks.

### CODE REVIEW

File: cmd/serv.go [hunk 1]
```diff
@@ -103,12 +103,12 @@ func fail(ctx context.Context, userMessage, logMsgFmt string, args ...any) error
 	// There appears to be a chance to cause a zombie process and failure to read the Exit status
 	// if nothing is outputted on stdout.
 	_, _ = fmt.Fprintln(os.Stdout, "")
-	_, _ = fmt.Fprintln(os.Stderr, "Gitea:", userMessage)
```
Comment (lunny) [inline cmd/serv.go]: unrelated changes.

Comment (pulltheflower) [reply cmd/serv.go]: Sorry, removed.

**Review by KN4CK3R — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by KN4CK3R** [comment]
CI fail is related.

File: routers/api/v1/utils/hook.go [hunk 2]
```diff
@@ -164,13 +165,20 @@ func addHook(ctx *context.APIContext, form *api.CreateHookOption, ownerID, repoI
 	if len(form.Events) == 0 {
 		form.Events = []string{"push"}
 	}
+
+	isSystemWebhook, err := strconv.ParseBool(form.Config["is_system_webhook"])
+	if err != nil {
+		ctx.Error(http.StatusUnprocessableEntity, "", "Invalid is_system_webhook value")
```
Comment (wxiaoguang) [inline routers/api/v1/utils/hook.go]: What is `is_system_webhook`? I didn't see any document for it.

And I do not think `err != nil` is right here, because `strconv.ParseBool("")` returns an error: is `is_system_webhook` optional?

The last question, what is `form.Config`? Is it from a JSON? Is there a chance to use "bool" type?

**Review by wxiaoguang — DISMISSED** [review]
Need more improvements

File: routers/api/v1/utils/hook.go (hunk 2 above)
Comment (pulltheflower) [reply routers/api/v1/utils/hook.go]: > What is is_system_webhook? I didn't see any document for it.

I haven't add the document about `is_system_webhook`, but I noticed that when we create system hooks in the gitea admin page, it will write the `is_system_webhook` column in `webhook` table.
image
And when we create webhooks via `/api/v1/admin/hooks` API, we can not choose what type of webhook to create. We can only create `default_webhook`. So I want add this field to this API.

>And I do not think err != nil is right here, because strconv.ParseBool("") returns an error: is is_system_webhook optional?

Thanks, You are right. I have optimize the code .

>The last question, what is form.Config? Is it from a JSON? Is there a chance to use "bool" type?

Yes, it's a `map[string]string`, if we change it to `map[string]any`, a lot of codes should be change.

**Comment by wxiaoguang** [comment]
Dismiss my change request, but I sill have the question about whether it (`is_system_webhook)` needs to be documented somewhere.

**Comment by delvh** [comment]
@wxiaoguang is this PR ready to be merged or not?

**Comment by wxiaoguang** [comment]
> @wxiaoguang is this PR ready to be merged or not?

Just nit, no block from my side.
